src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs

```diff
@@ -119,16 +119,26 @@ public async Task<IActionResult> Index(WorkflowTypeIndexOptions options, PagerPa
 
             var count = await query.CountAsync();
 
-            var workflowTypes = await query
+            var workflowTypes = (await query
                 .Skip(pager.GetStartIndex())
                 .Take(pager.PageSize)
-                .ListAsync();
+                .ListAsync())
+                .ToList();
 
             var workflowTypeIds = workflowTypes.Select(x => x.WorkflowTypeId).ToList();
-            var workflowGroups = (await _session.QueryIndex<WorkflowIndex>(x => x.WorkflowTypeId.IsIn(workflowTypeIds))
-                .ListAsync())
-                .GroupBy(x => x.WorkflowTypeId)
-                .ToDictionary(x => x.Key);
+
+            // Changing the workflowTypeIds list to array is needed to be able to modify the original list while going
+            // through it.
+            foreach (var workflowTypeId in workflowTypeIds.ToArray())
+            {
+                var workflowInstance = await _session.QueryIndex<WorkflowIndex>(index =>
+                        index.WorkflowTypeId == workflowTypeId)
+                    .FirstOrDefaultAsync();
```

Comment:
This could be `CountAsync()` as we don't need to retrieve the record from the database we just want to know if it exists.

Reply:
You are right we return unnecessary data, but it is still faster to return at the first instance than going through the table for counting all occurrences.